test/retained_topic_map.hpp

```diff
@@ -280,8 +303,9 @@ class retained_topic_map {
     // Dump debug information
     template<typename Output>
     void dump(Output &out) {
-        for (auto const& i : map) {
-            out << i.first.first << " " << i.first.second << " " << (i.second.value ? "init" : "-") << " " << i.second.count << std::endl;
+        auto const &direct_index = map.template get<direct_index_tag>();
+        for (auto const& i: direct_index) {
```

Comment:
```suggestion
        for (auto const& i : direct_index) {
```

Reply:
Wait a moment, because i was already updating according to coding guidelines